openedx/adg/lms/webinars/admin.py

```diff
@@ -65,58 +68,99 @@ class WebinarAdmin(WebinarAdminBase):
 
     form = WebinarForm
 
+    def __init__(self, model, admin_site):
+        """
+        Extend constructor to create instance variable for old webinar state to be stored in.
+
+        When an existing webinar is updated, its old state is lost after the ModelAdmin's `save_model` method is
+        executed. We want to store the old state before it is overridden so that it can be accessed in `save_related`
+        method, which is executed after `save_model` method.
+        """
+        super().__init__(model, admin_site)
+        self.old_webinar = None
+
     def get_fields(self, request, obj=None):
         """
         Override `get_fields` to dynamically set fields to be rendered.
         """
         fields = super().get_fields(request, obj)
         if not obj:
-            fields.remove('send_update_emails_to_registrants')
+            fields.remove(SEND_UPDATE_EMAILS_FIELD)
         return fields
 
+    def save_model(self, request, obj, form, change):
+        """
+        Extension of `save_model` to capture the old webinar state prior to updation
+        """
+        if change:
+            self.old_webinar = Webinar.objects.get(id=obj.id)
+
+        super().save_model(request, obj, form, change)
+
     def save_related(self, request, form, formsets, change):
         """
-        Extension of save_related for webinar to send emails when object is created or modified.
+        Extension of `save_related` for webinar to send emails when webinar is created or modified.
         """
+        new_members = []
+        removed_members = []
         if change and any(field in form.changed_data for field in ['co_hosts', 'presenter', 'panelists']):
-            update_webinar_team_registrations(form)
+            # The below method `get_newly_added_and_removed_team_members` must be called before
+            # `super().save_related(...)` for correct results.
```

Comment:
That actually makes sense. And I feel for you for all the trouble you had to go through 😭 

Is there any way we could reflect this in a unit test and add the explanation there? 
Barring that, can we add a line break above the comments.

Reply:
No, I meant adding a newline above the comment. It looks a little cluttered.

Reply:
Also, if it cannot be captured in a unit test, let's go ahead as it is.

Reply:
Yes, and above the comment is a conditional. We don't usually leave an empty line right after a conditional, and it would lead to inconcsistency. That is the code:

```
        if change and any(field in form.changed_data for field in ['co_hosts', 'presenter', 'panelists']):
            # The below method `get_newly_added_and_removed_team_members` must be called before
            # `super().save_related(...)` for correct results.
            new_members, removed_members = get_newly_added_and_removed_team_members(form, self.old_webinar)
```

Reply:
I understand.